### PR TITLE
Low fi landscape profile selection approach-1

### PR DESCRIPTION
**Pull request by nikitamarysolomanpvt**
## Explanation 
Duplicate of #611
This PR contain UI changes for landscape mode only. Immediately after this portrait mode UI update PR will be created. This PR contains approach where no custom libraries are used to controlling span count. Have integer files for different display metric to control span count.

## Mock Link

https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/4ae1ae9d-9ae9-4cd3-a15a-6c2a7a23b8be/PC-MP-Profile-Created-Visible-
## Screenshot 
440dpi
hdpi
mdpi

## Checklist
- [x] Add Administration Controls
- [x] App Language
- [x] Test cases

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by rt4914** [comment]
@nikitamarysolomanpvt Unassigning myself, as I think my part in this PR was to communicate information to Ben. If you still want me to review this PR please assign it back to me. Thanks.

**Comment by nikitamarysolomanpvt** [comment]
@BenHenning PTAL

File: app/src/main/java/org/oppia/app/profile/ProfileChooserFragmentPresenter.kt
Comment (BenHenning) [inline app/src/main/java/org/oppia/app/profile/ProfileChooserFragmentPresenter.kt]: We should be ellipsizing this in the UI, e.g. via TextView's ellipsize method. That also performs smart ellipsizing to ensure things aren't broken awkwardly (more relevant for sentences, but still a good idea to generally use).

File: app/src/main/res/drawable/ic_language_icon_grey_24dp.xml
Comment (BenHenning) [inline app/src/main/res/drawable/ic_language_icon_grey_24dp.xml:1]: Don't we already have a language & settings icons on the profile chooser? I only see landscape being changed--why aren't these icons also being used in portrait?

File: app/src/main/res/layout-land/profile_chooser_fragment.xml
Comment (BenHenning) [inline app/src/main/res/layout-land/profile_chooser_fragment.xml:16]: Do we really want part of the view to remain fixed, or should the whole screen scroll?

File: app/src/main/res/layout-land/profile_chooser_fragment.xml
Comment (BenHenning) [inline app/src/main/res/layout-land/profile_chooser_fragment.xml]: Prefer avoiding the nested layouting given that we are using a ConstraintLayout (prefer aligning with constraints, instead).

File: app/src/main/res/layout-land/profile_chooser_fragment.xml
Comment (BenHenning) [inline app/src/main/res/layout-land/profile_chooser_fragment.xml]: Ditto here--given that the outer layout is a ConstraintLayout, prefer to lay out these views with constraints and avoid the extra layering.

File: app/src/main/res/layout/profile_chooser_add_view.xml
Comment (BenHenning) [inline app/src/main/res/layout/profile_chooser_add_view.xml]: Why this change?

File: app/src/main/res/layout/profile_chooser_profile_view.xml
Comment (BenHenning) [inline app/src/main/res/layout/profile_chooser_profile_view.xml:34]: Why not keep the data-binding? If you ellipsize here, you can keep the data-binding.

File: app/src/main/res/values-hdpi/dimens.xml
Comment (BenHenning) [inline app/src/main/res/values-hdpi/dimens.xml]: Nit: ``profile_chooser_span_count``

File: app/src/main/res/values-sw480dp/dimens.xml
Comment (BenHenning) [inline app/src/main/res/values-sw480dp/dimens.xml]: Should we be handling tablet layouts here? I suggest avoiding that for now and deferring to whoever implements tablet support holistically (we may not even use the same exact layout for tablets).

File: app/src/main/res/values-hdpi/dimens.xml
Comment (BenHenning) [inline app/src/main/res/values-hdpi/dimens.xml]: Do we want to use a different span count for portrait versus landscape?

File: app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt]: Please include all qualifiers.

Also, maybe you should move this to be in app/src/test by itself. It's not actually a sharedTest (e.g. it won't pass on Espresso, and we don't ever expect it to pass on Espresso).

This is more or less a change detector test, but it seems valuable to have.

File: app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt]: Instead of testing the dimension directly, can you test that the wrapping is correct? E.g. that if you populate the layout that the recycler view is actually using a GridLayoutManager with the correct span count for each density qualifier case?

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @nikitamarysolomanpvt. I had some comments, but overall I find this approach *much* cleaner and easier to understand.

File: app/src/main/java/org/oppia/app/profile/ProfileChooserFragmentPresenter.kt
Comment (nikitamarysolomanpvt) [reply app/src/main/java/org/oppia/app/profile/ProfileChooserFragmentPresenter.kt]: @BenHenning  So lets keep 2 as max line and remove max length for the same ... Coz this was a requirement from designer doc.

File: app/src/main/res/drawable/ic_language_icon_grey_24dp.xml
Comment (nikitamarysolomanpvt) [reply app/src/main/res/drawable/ic_language_icon_grey_24dp.xml:1]: @BenHenning No it was not there in portrait development ,as mentioned in the previous PR #611 with custom adapter, portrait  fixes are yet to do. This PR focus only on landscape. Would be creating immediately after this PR fixes of portrait.

File: app/src/main/res/layout-land/profile_chooser_fragment.xml
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout-land/profile_chooser_fragment.xml:16]: @BenHenning Except the bottom portion we want the remaining to scroll PTAL https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/1fc109c3-31f3-4743-bd3b-76be39c0049a/PC-SP-Profile-Chooser-PIN-Available-

File: app/src/main/res/layout/profile_chooser_add_view.xml
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout/profile_chooser_add_view.xml]: As per mock and also it was looking better when we applied this limit.. As you suggest i can remove its.

File: app/src/main/res/layout/profile_chooser_profile_view.xml
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout/profile_chooser_profile_view.xml:34]: @BenHenning Because of that ellipsis not working with max limit. Will remove this as you want me to remove that max limit.

File: app/src/main/res/values-hdpi/dimens.xml
Comment (nikitamarysolomanpvt) [reply app/src/main/res/values-hdpi/dimens.xml]: Done

File: app/src/main/res/values-sw480dp/dimens.xml
Comment (nikitamarysolomanpvt) [reply app/src/main/res/values-sw480dp/dimens.xml]: Removed.

File: app/src/main/res/values-hdpi/dimens.xml
Comment (nikitamarysolomanpvt) [reply app/src/main/res/values-hdpi/dimens.xml]: @BenHenning Ya it's advised to use 2 as span count for portrait mode.

File: app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (nikitamarysolomanpvt) [reply app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt]: @BenHenning You mean espresso test?

File: app/src/main/res/layout/profile_chooser_profile_view.xml
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout/profile_chooser_profile_view.xml:34]: Done.

File: app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (nikitamarysolomanpvt) [reply app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt]: Done.

File: app/src/main/res/layout/profile_chooser_add_view.xml
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout/profile_chooser_add_view.xml]: Changed to wrap_content

File: app/src/main/java/org/oppia/app/profile/ProfileChooserFragmentPresenter.kt
Comment (nikitamarysolomanpvt) [reply app/src/main/java/org/oppia/app/profile/ProfileChooserFragmentPresenter.kt]: Removed the above code and changed to 
maxLines =2
ellipsize=end
in xml with databinding.

File: app/src/main/res/layout-land/profile_chooser_fragment.xml
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout-land/profile_chooser_fragment.xml]: Done.

File: app/src/main/res/layout-land/profile_chooser_fragment.xml
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout-land/profile_chooser_fragment.xml]: Done.

File: app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (BenHenning) [reply app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt]: No, I'm suggesting that with Robolectric you retrieve the recycler view, retrieve its GridLayoutManager, and verify that the span count is equal to this value for each test instead of checking the dimension correctly. As it stands, it's possible for the column count to be wrong and yet these tests still pass (e.g. by improperly initializing the GridLayoutManager). Does that help clarify?

File: app/src/main/res/values-hdpi/dimens.xml
Comment (BenHenning) [reply app/src/main/res/values-hdpi/dimens.xml]: For all display densities? Can you make sure the tests verify this portrait vs. landscape span count? It didn't seem like they did before.

File: app/src/main/res/layout-land/profile_chooser_fragment.xml
Comment (BenHenning) [reply app/src/main/res/layout-land/profile_chooser_fragment.xml:16]: Hmm I see. If this works okay, let's keep it as-is. I'm always a bit hesitant to used NestedScrollView with nested RecyclerView since the scrolling can get a bit wonky, but if this layout is simple enough then let's not change anything. One alternative approach would have been to just put everything except the controls in the recycler view, but let's not make that change if it isn't necessary.

Thanks for clarifying.

File: app/src/main/res/drawable/ic_language_icon_grey_24dp.xml
Comment (BenHenning) [reply app/src/main/res/drawable/ic_language_icon_grey_24dp.xml:1]: Ack thanks. This might also address one of my later comments about testing portrait. I defer to you on whether you want to make sure the span count is arranged correctly for portrait in this PR or the next, but please make sure it is handled & tested.

File: app/src/main/java/org/oppia/app/profile/ProfileChooserFragmentPresenter.kt
Comment (BenHenning) [inline app/src/main/java/org/oppia/app/profile/ProfileChooserFragmentPresenter.kt]: This constant appears unused, please remove it.

File: app/src/main/res/layout-land/profile_chooser_fragment.xml
Comment (BenHenning) [inline app/src/main/res/layout-land/profile_chooser_fragment.xml:19]: Color should be named ``profile_chooser_background`` for consistency.

File: app/src/main/res/layout-land/profile_chooser_fragment.xml
Comment (BenHenning) [inline app/src/main/res/layout-land/profile_chooser_fragment.xml:35]: Ditto here.

File: app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (BenHenning) [inline app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt]: Perhaps:

```kotlin
/** Tests for ensuring [ProfileChooserFragment] uses the correct column count for profiles based on display density. */
```

(might need to be wrapped on multiple lines).

File: app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (BenHenning) [reply app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt]: This appears only half-complete. You moved the file, but didn't add the missing tests for remaining qualifies. Please also resolve that.

**Review by BenHenning — APPROVED** [review]
Thanks @nikitamarysolomanpvt. The solution LGTM, but please fully resolve the remaining comments before merging.

File: app/src/main/res/layout-land/profile_chooser_fragment.xml
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout-land/profile_chooser_fragment.xml:16]: @BenHenning Okay. As if now i'am not changing. If you still want me to change after reviewing the app changes, let me know.

File: app/src/main/res/values-hdpi/dimens.xml
Comment (nikitamarysolomanpvt) [reply app/src/main/res/values-hdpi/dimens.xml]: @BenHenning Ya for all display densities span count 2 in portrait mode. If you want me to change this let me know... It will be addressed in PR of portrait mode.

File: app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (nikitamarysolomanpvt) [reply app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt]: > This appears only half-complete. You moved the file, but didn't add the missing tests for remaining qualifies. Please also resolve that.

Could you please elobrate what is the "missing tests for remaining qualifies".

File: app/src/main/java/org/oppia/app/profile/ProfileChooserFragmentPresenter.kt
Comment (nikitamarysolomanpvt) [reply app/src/main/java/org/oppia/app/profile/ProfileChooserFragmentPresenter.kt]: Done. Removed.

File: app/src/main/res/drawable/ic_language_icon_grey_24dp.xml
Comment (nikitamarysolomanpvt) [reply app/src/main/res/drawable/ic_language_icon_grey_24dp.xml:1]: Okay

Comment (nikitamarysolomanpvt) [reply app/src/main/res/drawable/ic_language_icon_grey_24dp.xml:1]: > Okay will be adding them in upcoming PR of portrait mode.

Comment (nikitamarysolomanpvt) [reply app/src/main/res/drawable/ic_language_icon_grey_24dp.xml:1]: @BenHenning Okay ,
will be adding them in upcoming PR of portrait mode.

File: app/src/main/res/layout-land/profile_chooser_fragment.xml
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout-land/profile_chooser_fragment.xml:19]: @BenHenning For consistency we introduced camel-casing, unless its shades of different colors. PTAL on https://github.com/oppia/oppia-android/blob/develop/app/src/main/res/values/colors.xml

File: app/src/main/res/layout-land/profile_chooser_fragment.xml
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout-land/profile_chooser_fragment.xml:35]: Ditto here.

File: app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (nikitamarysolomanpvt) [reply app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt]: Done.

File: app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (nikitamarysolomanpvt) [reply app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt]: Done.

File: app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (nikitamarysolomanpvt) [reply app/src/sharedTest/java/org/oppia/app/testing/ProfileChooserSpanTest.kt]: Done.

File: app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (rt4914) [inline app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt]: Make these private

File: app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (rt4914) [inline app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt:136]: Add blank space above this.

File: app/src/main/java/org/oppia/app/profile/ProfileActivityPresenter.kt
Comment (rt4914) [inline app/src/main/java/org/oppia/app/profile/ProfileActivityPresenter.kt]: nit:
```
ProfileChooserFragment(),
ProfileChooserFragmentTestActivity.TAG_PROFILE_CHOOSER_FRAGMENT
```

**Review by rt4914 — APPROVED** [review]
This looks really great.

File: app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (nikitamarysolomanpvt) [reply app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt]: Done

File: app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (nikitamarysolomanpvt) [reply app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt:136]: Done.

File: app/src/main/java/org/oppia/app/profile/ProfileActivityPresenter.kt
Comment (nikitamarysolomanpvt) [reply app/src/main/java/org/oppia/app/profile/ProfileActivityPresenter.kt]: Updated to
```
    activity.supportFragmentManager.beginTransaction().add(
        R.id.profile_chooser_fragment_placeholder,
        ProfileChooserFragment(), ProfileChooserFragmentTestActivity.TAG_PROFILE_CHOOSER_FRAGMENT
      ).commitNow()
```

**Comment by rt4914** [comment]
@nikitamarysolomanpvt I have already approved this PR.

**Comment by nikitamarysolomanpvt** [comment]
I had updated code with develop, resolved conflicts and did your
suggestions hence assigned you again. Thank you @Rajat Talesra
<rajattalesra4914@gmail.com>

**Review by rt4914 — APPROVED** [review]
Reviewed again. LGTM

File: app/src/main/res/layout-land/profile_chooser_fragment.xml
Comment (BenHenning) [reply app/src/main/res/layout-land/profile_chooser_fragment.xml:16]: Resolving since we're keeping this solution. Thanks.

File: app/src/main/res/layout-land/profile_chooser_fragment.xml
Comment (BenHenning) [reply app/src/main/res/layout-land/profile_chooser_fragment.xml:19]: Ack, similarly let's keep this for now. Later, we should make everything consistent.

File: app/src/main/java/org/oppia/app/testing/ProfileChooserFragmentTestActivity.kt
Comment (BenHenning) [inline app/src/main/java/org/oppia/app/testing/ProfileChooserFragmentTestActivity.kt:11]: Since this test activity is for verifying ProfileChooserFragment, we shouldn't use ProfileActivityPresenter (that's essentially the same as using ProfileActivity). Instead, we should inflate and add ProfileChooserFragment to a test layout only used by this test activity.

File: app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (BenHenning) [inline app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt:28]: Nit: ``Reference document: https://developer.android.com/reference/androidx/test/core/app/ActivityScenario``

File: app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (BenHenning) [inline app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt:24]: Nit: Should be ``TAG_PROFILE_CHOOSER_FRAGMENT_RECYCLER_VIEW`` (recycler view as two words) per https://developer.android.com/kotlin/style-guide#camel_case.

File: app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (BenHenning) [inline app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt:98]: Some parts of this test name could be improved, e.g.:

```kotlin
testProfileChooserFragmentRecyclerView_landscape_ldpi_hasCorrectSpanCount
```

Please apply this test pattern for each of the following tests.

File: app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (BenHenning) [inline app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt:44]: This fragment isn't technically the one you're launching later since you're initializing an activity in getProfileChooserTestActivity, then shutting it down. Instead, you should avoid caching the activity, fragment, and recycler view, and instead retrieve it locally, e.g.:

```kotlin
  @Test
  @Config(qualifiers = "land-mdpi")
  fun testProfileChooserFragmentRecyclerView_landscape_mdpi_hasCorrectSpanCount() {
    launch(ProfileChooserFragmentTestActivity::class.java).use { activity ->
      assertThat(getProfileRecyclerViewGridLayoutManager(acitivty).spanCount).isEqualTo(3)
    }
  }

  private fun getProfileRecyclerViewGridLayoutManager(activity: Activity): GridLayoutManager {
    return getProfileRecyclerView(activity).layoutManager as GridLayoutManager
  }

  private fun getProfileRecyclerView(activity: Activity): RecyclerView {
    return getProfileChooserFragment(activity).view?.findViewWithTag<View>(TAG_PROFILE_CHOOSER_FRAGMENT_RECYCLERVIEW)!! as RecyclerView
  }

  private fun getProfileChooserFragment(activity: Activity): ProfileChooserFragment {
    return activity.supportFragmentManager.findFragmentByTag(ProfileChooserFragmentTestActivity.TAG_PROFILE_CHOOSER_FRAGMENT) as ProfileChooserFragment
  }
```

File: app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (BenHenning) [inline app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt:68]: These dimension tests can be removed now that you have ``spanCount`` versions of them.

**Review by BenHenning — APPROVED** [review]
Thanks @nikitamarysolomanpvt -- I find the span count tests much clearer for correctness. Once you resolve my open comments, please feel free to submit.

File: app/src/main/java/org/oppia/app/testing/ProfileChooserFragmentTestActivity.kt
Comment (nikitamarysolomanpvt) [reply app/src/main/java/org/oppia/app/testing/ProfileChooserFragmentTestActivity.kt:11]: @BenHenning You mean not to test the fragment and just add the fragment layout to activity and test?
by the way the fragment is  passed to the constructor of view model, hence i created dedicated view model and xml for this test.

File: app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (nikitamarysolomanpvt) [reply app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt:44]: @BenHenning if we are not using fragment then why we need getProfileChooserFragment?

File: app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (nikitamarysolomanpvt) [reply app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt:98]: Done.

File: app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (nikitamarysolomanpvt) [reply app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt:24]: Removed

File: app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (nikitamarysolomanpvt) [reply app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt:28]: Done.

File: app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (nikitamarysolomanpvt) [reply app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt:68]: Done.

File: app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt
Comment (BenHenning) [reply app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt:44]: Aren't we using the fragment in the above suggested code?

**Comment by nikitamarysolomanpvt** [comment]
Find final approach in #937

**Pull request closed without merging**
